extensions/objects/models/objects.py

```diff
@@ -511,13 +511,32 @@ def normalize(cls, raw):
         """
 
         def _validate_expression(expression):
+            """Validates the given expression.
+
+            Args:
+                expression: dict(str, *). The expression to be verified in the
+                    dict format.
+
+            Raises:
+                AssertionError: The specified expression is not in the correct
+                    format.
+            """
             assert isinstance(expression, dict)
```

Comment:
Maybe better to change these to `self.assertTrue(isinstance(expression, dict))`

Reply:
ditto below. I think the self.assert.... syntax is preferred. If not, please feel free to ignore this comment!

Reply:
Do we want to do this? Though this doesn't directly relate to this PR. I am fine either way :)

Reply:
I tried adding this but then realised that `self` is not passed in any of the methods. It needs few code changes at other relevant places as well. `assert` statements have been used both ways in the codebase. So, I think it's fine for now.

Reply:
Ah I see, thanks for checking. Let's leave it as is.